itemstack.lua

```diff
@@ -20,15 +20,27 @@ end
 function ItemStack:get_count() return self._count end
 --* `set_count(count)`: returns a boolean indicating whether the item was cleared
 --    `count`: number, unsigned 16 bit integer
-function ItemStack:set_count(count) self._count = count end
+function ItemStack:set_count(count)
+	if count > 0 and count <= 65535 then
+		self._count = count
+		return true
+	else
+		self:clear()
+		return false
+	end
+end
 --* `get_wear()`: returns tool wear (`0`-`65535`), `0` for non-tools.
 function ItemStack:get_wear() return self._wear end
 --* `set_wear(wear)`: returns boolean indicating whether item was cleared
 --    `wear`: number, unsigned 16 bit integer
 function ItemStack:set_wear(wear)
-	assert(wear <= 65535, "ItemStack:set_wear invalid wear value: "..tostring(wear))
-	wear = wear < 0 and -((-wear) % 65536) or wear
-	self._wear = math.max(0, wear < 0 and 65536 + wear or wear)
+	if wear <= 65535 then
+		self._wear = wear % 65536
+		return true
+	else
+		self.clear()
+		return false
+	end
```

Comment:
Is this updated engine 5.5 behavior? (yeah previous wear behavior was bit weird but kinda useful)

Just in case reminder link #7
Also for assertion link #64 as some technic mod tests depends on strict validation.

Reply:
> Is this updated engine 5.5 behavior? (yeah previous wear behavior was bit weird but kinda useful)

I'm not certain, I'll take a look. If I remember, the reference I was looking at just casts the value as uint16.

Reply:
It looks like this is how it's been done since at least 0.4.17.1

Reply:
Are you confusing this with "add_wear"?

Reply:
Ah actually it is same behavior but just like engine allows destroying items by setting invalid wear?
Should prob just add some simplified strict to throw errors, probably duping `assert` as `strict_assert` or something for easy use (and custom error behavior handling)

Reply:
It destroys the item if the wear gets above 65535. If the wear is negative, it just casts it to uint16. 

I'm not sure what you're talking about w.r.t. simplified strict mode or throwing errors?

https://github.com/minetest/minetest/blob/2d8eac4e0a609acf7a26e59141e6c684fdb546d0/src/script/lua_api/l_item.cpp#L128-L147
https://github.com/minetest/minetest/blob/2d8eac4e0a609acf7a26e59141e6c684fdb546d0/src/inventory.h#L185

Reply:
Now wondering why I've even done that as it only affects values > 65536 and those will kill stack anyway.... 🤷 well, looks way better now so just have to add assertion thing.
I think I could try to look into that a bit tomorrow, would be nice to get this merged soon anyway.

Reply:
> I'm not sure what you're talking about w.r.t. simplified strict mode or throwing errors?

Strict mode means that error is raised when something is done that does not seem clearly intended behavior like destroying tool stack by setting invalid value for it. That is also application specific thing as sometimes it is meant to be destroyed while sometimes not, would however like to have assertion near actual call that causes potentially unwanted behavior for this specific case.
Not sure how this case should be exactly handled, so far was handled with straight assertion probably just because that happened to be only use case I needed back then...

Reply:
Correct me if I'm wrong, but destroying the stack by setting the value above 65535 *is* the intended behavior. You used the tool too much, now it's broken. The assertion that was there previously was wrong.

Reply:
Which is why clarified with _"That is also application specific thing as sometimes it is meant to be destroyed while sometimes not"_.
True that it is kind of wrong and should probably have been handled different way in tests from beginning, use case for that assertion is with indestructible rechargeable power tools.

It could have been done better with more specific assertions in test set by verifying that stack is still valid but... well, that was not done... but then early error also proved to be useful as when tool stack got accidentally destroyed it eventually caused error on completely irrelevant part of mod and that's reason why _"would however like to have assertion near actual call that causes potentially unwanted behavior for this specific case"_

Reply:
Probably needs way to mark item def as destructible/indestructible through "side effects" and global configuration for default behavior... I think that would work best here.